src/topology/subset_properties.lean

```diff
@@ -339,25 +348,17 @@ end
 
 /-- Finite topological spaces are compact. -/
 @[priority 100] instance fintype.compact_space [fintype α] : compact_space α :=
-{ compact_univ := compact_of_finite set.finite_univ }
+{ compact_univ := set.finite_univ.compact }
 
 /-- The product of two compact spaces is compact. -/
 instance [compact_space α] [compact_space β] : compact_space (α × β) :=
-⟨begin
-  have A : compact (set.prod (univ : set α) (univ : set β)) :=
-    compact_prod univ univ compact_univ compact_univ,
-  have : set.prod (univ : set α) (univ : set β) = (univ : set (α × β)) := by simp,
-  rwa this at A,
-end⟩
+⟨by { rw ← univ_prod_univ, exact compact_univ.prod compact_univ }⟩
 
 /-- The disjoint union of two compact spaces is compact. -/
 instance [compact_space α] [compact_space β] : compact_space (α ⊕ β) :=
 ⟨begin
-  have A : compact (@sum.inl α β '' univ) := compact_image compact_univ continuous_inl,
-  have B : compact (@sum.inr α β '' univ) := compact_image compact_univ continuous_inr,
-  have C := compact_union_of_compact A B,
-  have : (@sum.inl α β '' univ) ∪ (@sum.inr α β '' univ) = univ := by ext; cases x; simp,
-  rwa this at C,
+  rw ← range_inl_union_range_inr,
+  exact (compact_range continuous_inl).union (compact_range continuous_inr)
 end⟩
```

Comment:
@sgouezel What do you think about this style? Is it readable?

Reply:
For a 2-lines proof, sure, this is perfect.